lib/streamlit/elements/json.py

```diff
@@ -70,6 +71,18 @@ def json(
 
         if not isinstance(body, str):
             try:
+                # Convert sets into lists, which render more nicely on the frontend
+                set_found = False
+                for key in body:
+                    if isinstance(body[key], set):
+                        if not set_found:
+                            # When set is found to prevent mutation of input body, we need to deepcopy it once
+                            # To avoid copying it multiple times we use set_found flag
+                            body = copy.deepcopy(body)
```

Comment:
I used `copy.copy(body)` to perform a shallow copy, and the tests still passed - can we use that instead?

Reply:
Yes, `copy.copy` works perfectly fine. It's changed 👍 

Previously I've done mistake by copying it in the following way `new_body = body`, because I was thinking that Python is shallowcopying by default, seems like it's not and I'm reading more about this 👀